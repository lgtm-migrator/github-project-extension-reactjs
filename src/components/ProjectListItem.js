import React from "react";
import PropTypes from 'prop-types';
import Colors from "../data/colors.json";
import styled from "styled-components";

const H3 = styled.div`
    color: #2096F3;
    font-size: 18px;
    font-weight: bold;
    flex: 1;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
`
const Subtext = styled.div`
    font-size: 12px;
    font-style: italic;
    color: #90A4AE;
    flex: 2;
    text-overflow: ellipsis;
    overflow: hidden;
    line-height: normal;
`
const Text = styled.div`
    font-size: 12px;
    flex: 1;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow-wrap: break-word;
`

const LanguageText = styled.div`
    padding-top: 5px;
    font-size: 10px;
    flex: 1;
`
const Item = styled.div`
    height: 115px;
    display: flex;
    flex-direction: column;
    padding-top: 10px;
    padding-left: 10px;
    border-width: 0px 0px 1px 0px;
    border-style: solid;
    border-color: #CCC;
    background-color: #FFFFFF;
    transition: background-color 0.5s ease;
    cursor: ${props => props.draggable? 'move' : 'pointer' };
    overflow: hidden;
    
    a {
        flex-direction: column;
        display: flex;
        height: 100%;
        color: rgba(0,0,0,.87);

        &:focus, &:hover {
            text-decoration: none;
        }
    }

    &:first-child {
        border-top-width: 0px;
        height: 130px;

        &:before {
            content: '${ props => props.draggable ? '': 'HIGHLIGHTED' }';
            ${ props => props.draggable ? '': 'border: 1px solid #000;' }
            ${ props => props.draggable ? '': 'background-color: #cfd8dc;' }
            border-radius: 7px;
            padding: 0 5px 5px 5px;
            font-weight: bold;
            height: 14px;
            width: 95px;
        }
    }

    &:hover {
        background-color: #BBDEFB;
    }
`;

const PillaMaThing = styled.span`
    position: relative;
    top: 1px;
    display: inline-block;
    width: 12px;
    height: 12px;
    border-radius: 50%;
    background-color: ${props => props.color};
    margin-right: 3px;
`;

/**
 * ProjectListItem
 * 
 * Component that displays repo row
 * 
 * @param {Object} repo, onClick, draggable
 */
function ProjectListItem ({ repo={}, onClick, draggable}) {
    const pillColor = Colors[repo.language];

    return(
        <Item key={repo.id} draggable={draggable} >
            <a href={repo.html_url} target="_blank">
                <H3>{repo.name}</H3>
                <Text>{repo.full_name}</Text>
                <Subtext>{repo.description}</Subtext>
                <LanguageText><div>{repo.language ? <PillaMaThing color={pillColor} /> : '' }{repo.language}</div></LanguageText>
            </a>
        </Item>
    );
}

ProjectListItem.propTypes = {
    onClick: PropTypes.func,
    draggable: PropTypes.bool,
    repo: PropTypes.shape({
        id: PropTypes.string,
        name: PropTypes.string,
        full_name: PropTypes.string,
        description: PropTypes.string,
        language: PropTypes.string,
    })
};

export default ProjectListItem;