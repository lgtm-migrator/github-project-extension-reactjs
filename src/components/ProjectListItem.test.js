import React from 'react';
import { shallow, mount } from 'enzyme';
import ProjectListItem from './ProjectListItem';
import 'jest-styled-components';
const repo = {
    id: '1',
    name: 'Repo',
    full_name: 'Some Repo',
    description: 'Some repo description',
    language: 'Ruby',
};

describe('ProjectListItem ', () => {
    it('renders with no props', () => {
        shallow(<ProjectListItem />);
    });

    it('renders repo with matching snapshot', () => {
        const rendered = mount(<ProjectListItem repo={repo} />);
        expect(rendered).toMatchSnapshot();
    });

    it('changes to the correct pill color', () => {
        const rendered = shallow(<ProjectListItem repo={repo} />);
        let element = rendered.find('[color]');
        expect(element.props().color).toBe("#701516")
    });
});