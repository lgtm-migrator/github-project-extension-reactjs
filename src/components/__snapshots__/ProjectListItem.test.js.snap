// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectListItem  renders repo with matching snapshot 1`] = `
.c1 {
  color: #2096F3;
  font-size: 18px;
  font-weight: bold;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c3 {
  font-size: 12px;
  font-style: italic;
  color: #90A4AE;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  text-overflow: ellipsis;
  overflow: hidden;
  line-height: normal;
}

.c2 {
  font-size: 12px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow-wrap: break-word;
}

.c4 {
  padding-top: 5px;
  font-size: 10px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 {
  height: 115px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 10px;
  padding-left: 10px;
  border-width: 0px 0px 1px 0px;
  border-style: solid;
  border-color: #CCC;
  background-color: #FFFFFF;
  -webkit-transition: background-color 0.5s ease;
  transition: background-color 0.5s ease;
  cursor: pointer;
  overflow: hidden;
}

.c0 a {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  color: rgba(0,0,0,.87);
}

.c0 a:focus,
.c0 a:hover {
  text-decoration: none;
}

.c0:first-child {
  border-top-width: 0px;
  height: 130px;
}

.c0:first-child:before {
  content: 'HIGHLIGHTED';
  border: 1px solid #000;
  background-color: #cfd8dc;
  border-radius: 7px;
  padding: 0 5px 5px 5px;
  font-weight: bold;
  height: 14px;
  width: 95px;
}

.c0:hover {
  background-color: #BBDEFB;
}

.c5 {
  position: relative;
  top: 1px;
  display: inline-block;
  width: 12px;
  height: 12px;
  border-radius: 50%;
  background-color: #701516;
  margin-right: 3px;
}

<ProjectListItem
  repo={
    Object {
      "description": "Some repo description",
      "full_name": "Some Repo",
      "id": "1",
      "language": "Ruby",
      "name": "Repo",
    }
  }
>
  <styled.div
    key="1"
  >
    <div
      className="c0"
    >
      <a
        target="_blank"
      >
        <styled.div>
          <div
            className="c1"
          >
            Repo
          </div>
        </styled.div>
        <styled.div>
          <div
            className="c2"
          >
            Some Repo
          </div>
        </styled.div>
        <styled.div>
          <div
            className="c3"
          >
            Some repo description
          </div>
        </styled.div>
        <styled.div>
          <div
            className="c4"
          >
            <div>
              <styled.span
                color="#701516"
              >
                <span
                  className="c5"
                  color="#701516"
                />
              </styled.span>
              Ruby
            </div>
          </div>
        </styled.div>
      </a>
    </div>
  </styled.div>
</ProjectListItem>
`;
